src/profile-logic/marker-data.js

```diff
@@ -0,0 +1,318 @@
+        continue;
+      }
+      if (
+        typeof data.category === 'string' &&
+        data.category.toLowerCase().includes(lowerCaseSearchString)
+      ) {
+        // Match UserTiming's name.
+        newMarkers.push(marker);
+        continue;
+      }
+    }
+  }
+  return newMarkers;
+}
+
+/**
+ * This function takes a marker that packs in a marker payload into the string of the
+ * name. This extracts that and turns it into a payload.
+ */
+export function extractMarkerDataFromName(
+  markers: MarkersTable,
+  stringTable: UniqueStringArray
+): MarkersTable {
+  const newMarkers: MarkersTable = {
+    data: markers.data.slice(),
+    name: markers.name.slice(),
+    time: markers.time.slice(),
+    length: markers.length,
+  };
+
+  // Match: "Bailout_MonitorTypes after add on line 1013 of self-hosted:1008"
+  // Match: "Bailout_TypeBarrierO at jumptarget on line 1490 of resource://devtools/shared/base-loader.js -> resource://devtools/client/shared/vendor/immutable.js:1484"
+  const bailoutRegex =
+    // Capture groups:
+    //       type   afterAt    where        bailoutLine  script functionLine
+    //        ↓     ↓          ↓                  ↓        ↓    ↓
+    /^Bailout_(\w+) (after|at) ([\w _-]+) on line (\d+) of (.*):(\d+)$/;
+
+  // Match: "Invalidate resource://devtools/shared/base-loader.js -> resource://devtools/client/shared/vendor/immutable.js:3662"
+  // Match: "Invalidate self-hosted:4032"
+  const invalidateRegex =
+    // Capture groups:
+    //         url    line
+    //           ↓    ↓
+    /^Invalidate (.*):(\d+)$/;
+
+  for (let markerIndex = 0; markerIndex < markers.length; markerIndex++) {
+    const nameIndex = markers.name[markerIndex];
+    const time = markers.time[markerIndex];
+    const name = stringTable.getString(nameIndex);
+    let matchFound = false;
+    if (name.startsWith('Bailout_')) {
+      matchFound = true;
+      const match = name.match(bailoutRegex);
+      if (!match) {
+        console.error(`Could not match regex for bailout: "${name}"`);
+      } else {
+        const [
+          ,
+          type,
+          afterAt,
+          where,
+          bailoutLine,
+          script,
+          functionLine,
+        ] = match;
+        newMarkers.name[markerIndex] = nameIndex;
+        newMarkers.data[markerIndex] = ({
+          type: 'Bailout',
+          bailoutType: type,
+          where: afterAt + ' ' + where,
+          script: script,
+          bailoutLine: +bailoutLine,
+          functionLine: +functionLine,
+          startTime: time,
+          endTime: time,
+        }: BailoutPayload);
+      }
+    } else if (name.startsWith('Invalidate ')) {
+      matchFound = true;
+      const match = name.match(invalidateRegex);
+      if (!match) {
+        console.error(`Could not match regex for bailout: "${name}"`);
+      } else {
+        const [, url, line] = match;
+        newMarkers.data[markerIndex] = {
+          type: 'Invalidation',
```

Comment:
same thing here... 
You completely removed the `name` change to keep the previous one, but the same comment applies.